inspirehep/utils/stats.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import operator
+
+
+def calculate_h_index(citations):
+    """
+    Calculate the h-index of a citation dictionary.
+
+    An author has h-index X if she has X papers with at least X citations each.
+    See: https://en.wikipedia.org/wiki/H-index.
+
+    :param citations: a dictionary in the format {recid: citation_count}
+    :return: h-index of the dictionary of citations.
+    """
+    sorted_citations = _sort_citations(citations)
+
+    h_index = 0
+    for i, citation in enumerate(sorted_citations):
+        if citation[1] > i:
+            h_index += 1
+        else:
+            break
+
+    return h_index
+
+
+def calculate_i10_index(citations):
+    """
+    Calculate the i10-index of a citation dictionary.
+
+    An author has i10-index X if she has X papers with at least 10 citations
+    each. See: https://en.wikipedia.org/wiki/H-index#i10-index
+
+    :param citations: a dictionary in the format {recid: citation_count}
+    :return: i10-index of the dictionary of citations.
+    """
+    sorted_citations = _sort_citations(citations)
+
+    i10_index = 0
+    for citation in sorted_citations:
```

Comment:
My intent here was to write the two algorithms as similar as possible so that one could tell at a glance the difference. But yeah, sorting here is useless (but N log N and N are not that distant when N ~ 1000...)